.github/workflows/release.yml

```diff
@@ -9,18 +9,38 @@ on:
         description: 'Tag to release'
         required: true
 
+permissions:
+  contents: read
+
 jobs:
-  release:
+  build-wheel:
+    name: Build wheel
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         if: ${{ github.event_name == 'release' }}
-      - uses: actions/checkout@v2
-        if: ${{ github.event_name == 'workflow_dispatch' }}
+      - uses: actions/checkout@v4
         with:
           ref: ${{ github.event.inputs.ref }}
-      - name: Create dist
-        run: make wheel
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+      - run: python -m pip install hatch
+      - name: Build package
+        run: hatch build
+      - uses: actions/upload-artifact@v3
```

Comment:
Not that it's wrong, just want to mention I don't love these upload/download artifacts steps. Why not just have a simple, single publish/release step?  
@psss Is the Trusted Publishers all set on PyPI?

Reply:
Most of these workflows are from some other projects I manage and it was important there because those wheels are arch dependent. For this project, indeed all of that is overkill and can be compressed, especially if we don't want to use [`hynek/build-and-inspect-python-package`](https://github.com/hynek/build-and-inspect-python-package). The only potential usecase I see is when we want to update `pyproject.toml` metadata field, which should be rare enough that it would be fine without.

Having a `release.yml` workflow setup in general though, I think it should be setup, but probably in another PR and have it more fine-tuned to the project's style of release